packages/replay/src/replay.ts

```diff
@@ -922,7 +922,7 @@ export class ReplayContainer implements ReplayContainerInterface {
     const transport = client && client.getTransport();
     const dsn = client?.getDsn();
 
-    if (!client || !scope || !transport || !dsn) {
+    if (!client || !scope || !transport || !dsn || !this.session || !this.session.sampled) {
```

Comment:
This should never be the case in this method but needed for ts and got rid of optional chaining